tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+"""BleBox light entities tests."""
+
+import logging
+
+import blebox_uniapi
+import pytest
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    ATTR_WHITE_VALUE,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_WHITE_VALUE,
+)
+from homeassistant.const import (
+    ATTR_SUPPORTED_FEATURES,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.util import color
+
+from .conftest import async_setup_entity, mock_feature
+
+from tests.async_mock import AsyncMock, PropertyMock
+
+ALL_LIGHT_FIXTURES = ["dimmer", "wlightbox_s", "wlightbox"]
+
+
+@pytest.fixture(name="dimmer")
+def dimmer_fixture():
+    """Return a default light entity mock."""
+
+    feature = mock_feature(
+        "lights",
+        blebox_uniapi.light.Light,
+        unique_id="BleBox-dimmerBox-1afe34e750b8-brightness",
+        full_name="dimmerBox-brightness",
+        device_class=None,
+        brightness=65,
+        is_on=True,
+        supports_color=False,
+        supports_white=False,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My dimmer")
+    type(product).model = PropertyMock(return_value="dimmerBox")
+    return (feature, "light.dimmerbox_brightness")
+
+
+async def test_dimmer_init(dimmer, hass, config):
+    """Test cover default state."""
+
+    _, entity_id = dimmer
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-dimmerBox-1afe34e750b8-brightness"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "dimmerBox-brightness"
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_BRIGHTNESS
+
+    assert state.attributes[ATTR_BRIGHTNESS] == 65
+    assert state.state == STATE_ON
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My dimmer"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "dimmerBox"
+    assert device.sw_version == "1.23"
+
+
+async def test_dimmer_update(dimmer, hass, config):
+    """Test light updating."""
+
+    feature_mock, entity_id = dimmer
+
+    def initial_update():
+        feature_mock.brightness = 53
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_BRIGHTNESS] == 53
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)